scheduler/views.py

```diff
@@ -348,3 +349,51 @@ def get_success_url(self):
         Redirect to the same page.
         """
         return reverse("planner_by_facility", kwargs=self.kwargs)
+
+
+class SendMessageToShiftHelpers(LoginRequiredMixin, FormView):
+    """
+    View processes the sending of an email to all shift helpers.
+    """
+
+    template_name = "helpdesk_single.html"
+    form_class = ShiftMessageToHelpersModelForm
+
+    def form_invalid(self, form):
+
+        messages.warning(self.request, _("The submitted data was invalid."))
+        return super(SendMessageToShiftHelpers, self).form_invalid(form)
+
+    def form_valid(self, form):
+        user = self.request.user
+        try:
+            user_account = UserAccount.objects.get(user=user)
+        except UserAccount.DoesNotExist:
+            messages.warning(self.request, _("User account does not exist."))
+            return super(SendMessageToShiftHelpers, self).form_valid(form)
+
+        shift = form.cleaned_data.get("shift")
+        if not is_facility_member(user_account.user, shift.facility):
+            messages.warning(
+                self.request, _("The user has no permissions to send emails!")
+            )
+            return super(SendMessageToShiftHelpers, self).form_valid(form)
+
+        sm = ShiftMessageToHelpers.objects.create(
+            message=form.cleaned_data.get("message"),
+            shift=shift,
+            sender=user_account,
+        )
+        for helper in shift.helpers.all():
+            if helper.user.email:
+                sm.recipients.add(helper)
+                sm.save()
```

Comment:
Correct me, if I'm wrong, but:

Could it be, this sends 'n' E-Mails to the first helper?
Here there is an iteration over "helpers" and each helpers email is added as recipient.
The "save" triggers "post_save" signal, which iterates all recipients.
So if 20 people signed up, the first account in list might get 20 Mails?

Wouldn't it be sufficient, to bulk-add all - before saving one time?

```suggestion
        sm.recipients.add(*shift.helpers.all())
        sm.save()
```

Reply:
+1 for not calling save n times, but it is not sending n messages to the first helper.

Reply:
Einmal...